staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/config/authentication.go

```diff
@@ -31,17 +31,22 @@ import (
 // rest.Config generated by the resolver.
 type AuthenticationInfoResolverWrapper func(AuthenticationInfoResolver) AuthenticationInfoResolver
 
-// AuthenticationInfoResolver builds rest.Config base on the server name.
+// AuthenticationInfoResolver builds rest.Config base on the server name and
+// the directRouting flag indicating whether the webhook server is routable
+// directly from apiserver's network environment.
+//
+// TODO(yguo0905): Remove the directRouting flag once the SSH tunnels that is
+// used for the communication from master to nodes get removed.
 type AuthenticationInfoResolver interface {
-	ClientConfigFor(server string) (*rest.Config, error)
+	ClientConfigFor(server string, directRouting bool) (*rest.Config, error)
```

Comment:
More than just the kube apiserver will use this. We shouldn't change the interface just to accommodate kube apiserver

Reply:
I think the new flag might be useful in other cases. Or do you mean the client of the AuthenticationInfoResolver should never care where/how the webhooks are deployed?

Reply:
Open to other suggestions. Given that we do have the ssh tunnels right now this doesn't seem completely wrong to me.

Reply:
since the API driving webhooks distinguishes between service name/namespace and url.URL, seems like it would be better to just make that the API here:

```
ClientConfig(server string) (*rest.Config, error)
ClientConfigForService(serviceName, serviceNamespace string) (*rest.Config, error)
```

kube-apiserver's override would only inject the proxy dialer for ClientConfigForService